### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by skejserjensen**
This PR primarily updates Delta Lake to [v0.22.0](https://crates.io/crates/deltalake-core/0.22.0) and Apache DataFusion to [v43.0.0](https://crates.io/crates/datafusion/43.0.0) and fixes all known breaking changes. New versions of Apache DataFusion usually have a few breaking changes, so updating to each new release ensures that the number of changes required does not accumulate.

Two breaking changes required more complex changes. The API for adding aggregates to Apache DataFusion has been changed, e.g., has `AggregateExpr` been removed,  thus `model_simple_aggregates.rs` has been changed to use the new API. Also `sysinfo` no longer removes killed processes when `System.refresh_all()` is called as `System.refresh_processes()` now has a `remove_dead_processes` parameter which `System.refresh_all()` passes `false` to. Thus, `System.refresh_processes_specifics()` is now explicitly used in `integration_test.rs` to ensure killed processes are removed and to provide the id of the process to update to not waste cycles updating all processes.

So the PR can be reviewed in parallel with #259 it is opened against `dev/single-delta-lake-type`, thus the target branch should be changed to `main` before it is merged.

### CODE REVIEW

**Review by CGodiksen — CHANGES_REQUESTED** [review]

File: crates/modelardb_storage/src/metadata/model_table_metadata.rs [hunk 1]
```diff
@@ -219,7 +219,10 @@ pub struct GeneratedColumn {
 mod test {
     use super::*;
 
-    use datafusion::arrow::datatypes::{DataType, Field, Schema};
+    use datafusion::{
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/metadata/model_table_metadata.rs:222]: Fixed.

File: crates/modelardb_storage/src/query/mod.rs [hunk 2]
```diff
@@ -31,22 +32,21 @@ pub(crate) mod normal_table;
 pub(crate) mod sorted_join_exec;
 
 /// The global sort order [`datafusion::datasource::physical_plan::parquet::ParquetExec`] guarantees
-/// for the segments it produces and that [`grid_exec::GridExec`] requires for the segments its
-/// receives as its input. It is guaranteed by
+/// for the segments it produces. It is guaranteed by
 /// [`datafusion::datasource::physical_plan::parquet::ParquetExec`] because the storage engine uses
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/query/mod.rs:36]: Fixed.

File: crates/modelardb_storage/src/query/mod.rs [hunk 3]
```diff
@@ -55,25 +55,42 @@ static QUERY_ORDER_SEGMENT: LazyLock<Vec<PhysicalSortExpr>> = LazyLock::new(|| {
             expr: Arc::new(Column::new("start_time", 2)),
             options: sort_options,
         },
-    ]
+    ];
+
+    LexOrdering::new(physical_sort_expr)
+});
+
+/// The global sort order that [`grid_exec::GridExec`] requires for the segments its receives as its
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/query/mod.rs:63]: Fixed.

File: crates/modelardb_storage/src/query/model_table.rs [hunk 4]
```diff
@@ -138,6 +140,17 @@ impl ModelTable {
     }
 }
 
+/// The implementation is not derived many instance variables does not implement [`fmt::Debug`].
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/query/model_table.rs:143]: Fixed.

File: crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs [hunk 5]
```diff
@@ -444,13 +334,18 @@ struct ModelCountAccumulator {
...
-                .unwrap()
-                .value(0);
+        let start_times = modelardb_types::value!(values, 2, TimestampArray);
+        let end_times = modelardb_types::value!(values, 3, TimestampArray);
+        let timestamps = modelardb_types::value!(values, 4, BinaryArray);
+
+        for row_index in 0..start_times.len() {
+            let start_time = start_times.value(row_index);
+            let end_time = end_times.value(row_index);
+            let timestamps = timestamps.value(row_index);
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs:344]: I kept the `timestamp` to keep the code consistent with the other query processing components such as `GridExec`.

File: crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs [hunk 6]
```diff
@@ -715,60 +460,28 @@ impl Accumulator for ModelMaxAccumulator {
...
+    /// Update the [`Accumulators`](Accumulator) state from `values`.
+    fn update_batch(&mut self, values: &[ArrayRef]) -> DataFusionResult<()> {
+        let model_type_ids = modelardb_types::value!(values, 1, UInt8Array);
+        let start_times = modelardb_types::value!(values, 2, TimestampArray);
+        let end_times = modelardb_types::value!(values, 3, TimestampArray);
+        let timestamps = modelardb_types::value!(values, 4, BinaryArray);
+        let min_values = modelardb_types::value!(values, 5, ValueArray);
+        let max_values = modelardb_types::value!(values, 6, ValueArray);
+
+        // Index order is swapped to keep naming consistent with Apache DataFusion and ModelaDB.
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs:480]: Fixed.

File: crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs [hunk 7]
```diff
@@ -715,60 +460,28 @@ impl Accumulator for ModelMaxAccumulator {
...
+        let residuals = modelardb_types::value!(values, 8, BinaryArray);
+        let values = modelardb_types::value!(values, 7, BinaryArray);
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs:482]: Fixed by renaming the method's parameter from `values` to `arrays` as it seems to make the code the least inconsistent. This should only require that we also rename `values` to `arrays` in `Accumulator.retract_batch()` if we ever implement it.

File: crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs [hunk 8]
```diff
@@ -865,61 +532,30 @@ impl Accumulator for ModelSumAccumulator {
...
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs:556]: See this [comment](https://github.com/ModelarData/ModelarDB-RS/pull/261#discussion_r1857392500).

**Review by CGodiksen — APPROVED** [review]

File: crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs [hunk 9]
```diff
@@ -176,281 +292,60 @@ fn try_new_aggregate_exec(
...
+
+        let model_aggregate_function_expr = match &name_with_column[..name_end] {
+            "count" => MODEL_AGGREGATE_COUNT.clone(),
+            "min" => MODEL_AGGREGATE_MIN.clone(),
+            "max" => MODEL_AGGREGATE_MAX.clone(),
+            "sum" => MODEL_AGGREGATE_SUM.clone(),
+            "avg" => MODEL_AGGREGATE_AVG.clone(),
+            name => {
+                return Err(DataFusionError::NotImplemented(format!(
+                    "Aggregate expression {name} is currently not supported.",
```
Comment (chrthomsen) [inline crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs:317]: I know this is not a new message, but should we remove "currently" which makes it sound somewhat like it soon will be supported

**Review by chrthomsen — APPROVED** [review]

File: crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs (hunk 9 above)
Comment (skejserjensen) [reply crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs]: Fixed.

**Pull request merged**